processor/src/operations/ext2_ops.rs

```diff
@@ -0,0 +1,78 @@
+use super::{AdviceProvider, ExecutionError, Felt, Process};
+
+// EXTENSION FIELD OPERATIONS
+// ================================================================================================
+
+impl<A> Process<A>
+where
+    A: AdviceProvider,
+{
+    // ARITHMETIC OPERATIONS
+    // --------------------------------------------------------------------------------------------
+    /// Gets the top four values from the stack [a1, a0, b1, b0], where a = (a1, a0) and
+    /// b = (b1, b0) are elements of the extension field, and outputs the product c = (c1, c0)
+    /// where c0 = b0 * a0 - 2 * b1 * a1 and c1 = (b0 + b1) * (a1 + a0) - b0 * a0. It pushes 0 to
+    /// the first and second positions on the stack, c1 and c2 to the third and fourth positions,
+    /// and leaves the rest of the stack unchanged.
+    pub(super) fn op_ext2mul(&mut self) -> Result<(), ExecutionError> {
+        let [a0, a1, b0, b1] = self.stack.get_top_word();
+        self.stack.set(0, b1);
+        self.stack.set(1, b0);
+        self.stack.set(2, (b0 + b1) * (a1 + a0) - b0 * a0);
+        self.stack.set(3, b0 * a0 - Felt::new(2) * b1 * a1);
+        self.stack.copy_state(4);
+        Ok(())
+    }
+}
+
+// TESTS
+// ================================================================================================
+
+#[cfg(test)]
+mod tests {
+    use super::{
+        super::{Felt, FieldElement, Operation, STACK_TOP_SIZE},
+        Process,
+    };
+    use rand_utils::rand_value;
+
+    // ARITHMETIC OPERATIONS
+    // --------------------------------------------------------------------------------------------
+
+    #[test]
+    fn op_ext2mul() {
+        // initialize the stack with a few values
+        let [a0, a1, b0, b1] = [rand_value(); 4];
+
+        let mut process = Process::new_dummy(&[a0, a1, b0, b1]);
+
+        // multiply the top two values
+        process.execute_op(Operation::Ext2Mul).unwrap();
+        let [a0, a1, b0, b1] = [Felt::new(a0), Felt::new(a1), Felt::new(b0), Felt::new(b1)];
+        let expected = build_expected(&[
+            b1,
+            b0,
+            (b0 + b1) * (a1 + a0) - b0 * a0,
+            b0 * a0 - Felt::new(2) * b1 * a1,
```

Comment:
I would actually perform multiplication using degree 2 extension field elements instead of doing it manually.